### PR TITLE
[18CZ] Fixed scrapping of trains

### LINKED ISSUE #4211: [18CZ] should not be able to purchase a scrapped train
NWB bought in BCB and scrapped both its `2a` trains

EKJ is able to buy in those `2a` trains from NWB on its turn, even though they have been scrapped

[test case](https://gist.github.com/benjaminxscott/e90674ef9a3f35d6b329aae33abb1e14)
edit: fixed json

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Needed to nil the owner

closes #4211

could break games, but probably very unlikely

### CODE REVIEW

**Comment by benjaminxscott** [comment]
Perhaps [remove_train](https://github.com/tobymao/18xx/blob/master/lib/engine/depot.rb#L71) could generically set owner to nil?

**Comment by Zwergenpunk** [comment]
you linked the depot. Thats the wrong one. We call the game.remove_train.

Yeah maybe, but I have no idea how much that breaks everything.

File: lib/engine/game/g_18_cz/game.rb [hunk 1]
```diff
@@ -2756,7 +2756,7 @@ def rust_trains!(train, entity)
             next unless rust?(t)
 
             rusted_trains << t.name
-            owners[t.owner.name] += 1
+            owners[t.owner.name] += 1 unless t.owner.nil?
```
Comment (tobymao) [inline lib/engine/game/g_18_cz/game.rb:2759]: if t.owner

Comment (Zwergenpunk) [reply lib/engine/game/g_18_cz/game.rb:2759]: When I remove .nil? the tests fail for some reason =(

Comment (Zwergenpunk) [reply lib/engine/game/g_18_cz/game.rb:2759]: Oh i am an idiot...

**Pull request merged**